clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntityFactory.java

```diff
@@ -100,44 +100,60 @@ public void create(final String identifier, final ServerSideConfiguration config
               + identifier + ": another client owns the maintenance lease");
-    } else {
-      try {
-        EntityRef<EhcacheClientEntity, UUID> ref = getEntityRef(identifier);
+    }
+
+    EntityRef<EhcacheClientEntity, UUID> ref = getEntityRef(identifier);
+    try {
+      while (true) {
+        ref.create(UUID.randomUUID());
         try {
-          while (true) {
-            ref.create(UUID.randomUUID());
-            try {
-              EhcacheClientEntity entity = ref.fetchEntity();
-              try {
-                entity.setTimeouts(entityTimeouts);
-                entity.configure(config);
-                return;
-              } finally {
-                entity.close();
-              }
-            } catch (ClusteredTierManagerConfigurationException e) {
-              try {
-                ref.destroy();
-              } catch (EntityNotFoundException f) {
-                //ignore
-              }
-              throw new EhcacheEntityCreationException("Unable to configure clustered tier manager for id " + identifier, e);
-            } catch (EntityNotFoundException e) {
-              //continue;
-            }
+          EhcacheClientEntity entity = ref.fetchEntity();
+          try {
+            entity.setTimeouts(entityTimeouts);
+            entity.configure(config);
+
+            break;
+          } finally {
+            entity.close();
           }
-        } catch (EntityNotProvidedException e) {
-          LOGGER.error("Unable to create clustered tier manager for id {}", identifier, e);
-          throw new AssertionError(e);
-        } catch (EntityVersionMismatchException e) {
-          LOGGER.error("Unable to create clustered tier manager for id {}", identifier, e);
-          throw new AssertionError(e);
-        }
-      } finally {
-        if (localMaintenance != null) {
-          localMaintenance.unlock();
+        } catch (ClusteredTierManagerConfigurationException e) {
+          try {
+            ref.destroy();
+          } catch (EntityNotFoundException f) {
+            //ignore
+          }
+          throw new EhcacheEntityCreationException("Unable to configure clustered tier manager for id " + identifier, e);
+        } catch (EntityNotFoundException e) {
+          //continue;
         }
       }
+    } catch (EntityNotProvidedException e) {
+      LOGGER.error("Unable to create clustered tier manager for id {}", identifier, e);
+      silentlyUnlock(localMaintenance, identifier);
+      throw new AssertionError(e);
+    } catch (EntityVersionMismatchException e) {
+      LOGGER.error("Unable to create clustered tier manager for id {}", identifier, e);
+      silentlyUnlock(localMaintenance, identifier);
+      throw new AssertionError(e);
+    } catch(EntityAlreadyExistsException e) {
+      silentlyUnlock(localMaintenance, identifier);
+      throw e;
+    } catch(EhcacheEntityCreationException e) {
```

Comment:
`EhcacheEntityCreationException` is a `RuntimeException` and since we do not have any specific handling, catching the higher type makes more sense and that is already done below.